### ISSUE TITLE
F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [d4rken-org/sdmaid-se#1232]
...as the Release APK has random suffixes

The F-Droid recipe https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/eu.darken.sdmse.yml#L19 is templated to have `-FOSS-RELEASE.apk` at the end

while https://github.com/d4rken-org/sdmaid-se/releases/tag/v1.1.0-beta0 has `-FOSS-BETA.apk`

ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/7125528127#L2933

maybe you want some tags skipped? eg. pickup only those tags that are not `beta` ?

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/ed1feb68f979fc9f222a344593429f983bdd5c6b

**Comment by licaon-kter** [comment]
Skipping betas for now: https://gitlab.com/fdroid/fdroiddata/-/commit/35c788654c8a0399ee3aaea4b00316a49fa59cc9

**Comment by d4rken** [comment]
Can beta releases be marked as beta on F-Droid?

I'd be fine to skip betas on f-droid.

Maybe only release versions on f-droid and all versions including pre-releases on @IzzySoft's repo. Thoughts?

**Comment by licaon-kter** [comment]
> Can beta releases be marked as beta on F-Droid?

not in an automated way, just manually

> I'd be fine to skip betas on f-droid.

great

**Comment by licaon-kter** [comment]
> Maybe only release versions on f-droid and all versions including pre-releases on IzzySoft's repo. Thoughts?

as you see fit

**Comment by IzzySoft** [comment]
> Maybe only release versions on f-droid and all versions including pre-releases on @IzzySoft's repo. Thoughts?

Sounds good to me! And there is precedence to this combination :stuck_out_tongue_winking_eye:

**Comment by licaon-kter** [comment]
A new fail: https://monitor.f-droid.org/builds/log/eu.darken.sdmse/10201000#site-footer

I'm building locally to diff `baseline.prof` via: https://github.com/obfusk/reproducible-apk-tools?tab=readme-ov-file#dump-baselinepy

```
$ diff -ru baseline.prof.dump_upstream baseline.prof.dump
--- baseline.prof.dump_upstream
+++ baseline.prof.dump
@@ -1,16 +1,16 @@
 entry=assets/dexopt/baseline.prof
 prof version=010 P
 num_dex_files=1
-uncompressed_data_size=14053
-compressed_data_size=2101
+uncompressed_data_size=14049
+compressed_data_size=2096
 dex_data_header 0
   profile_key='classes.dex'
   num_type_ids=314
-  hot_method_region_size=3268
+  hot_method_region_size=3264
   dex_checksum=0x4227e7c0
   num_method_ids=40517
 dex_data 0
-  num_hot_method_ids=817
+  num_hot_method_ids=816
   num_type_ids=314
   method_id=3285
   num_inline_caches=0
@@ -1478,8 +1478,6 @@
   num_inline_caches=0
   method_id=12561
   num_inline_caches=0
-  method_id=12562
-  num_inline_caches=0
   method_id=12577
   num_inline_caches=0
   method_id=12590
```
Then I rebuild again locally ... because past experiences... and... first try is successful, for some reason :shrug: 

Then used 3 VMs to rebuild... and all failed, again.

**Comment by IzzySoft** [comment]
I can confirm that btw: for v1.2.2 RB was broken in our builders at IzzyOnDroid as well – with about the same diff @licaon-kter posted above. Previous versions did not have that issue. Maybe we just got lucky with those then that we always hit the "success" path there. It would then be what we call a "flaky build". You can find [our build logs here](https://codeberg.org/IzzyOnDroid/rbtlog/src/branch/izzy/log/logs/eu.darken.sdmse.json). Will add a note here to set it up for auto-retry instead of having it marked "fail" next time.

Btw: Bluemusic is RB, too. Just added it to my verification builder now.

**Comment by d4rken** [comment]
I have no idea what this could be 🤷 

What's `RB`?

**Comment by licaon-kter** [comment]
Reproducible Build :)

**Comment by IzzySoft** [comment]
See:

* [Reproducible Builds](https://reproducible-builds.org/) (the RB project)
* [Reproducible Builds, special client support and more in our repo](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients) (the IzzyOnDroid announcement)
* [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds) (for app developers)

:wink:

**Comment by linsui** [comment]
https://gitlab.com/fdroid/fdroiddata/-/jobs/7550757742 It failed again. I'll just let the buildserver give it another try.

**Comment by linsui** [comment]
1.2.5-rc0 builds.

**Comment by licaon-kter** [comment]
1.2.8-rc0 does not :( https://monitor.f-droid.org/builds/log/eu.darken.sdmse/10208000#site-footer

and a build loop locally failed 10/10 times

can we reopen?

**Comment by d4rken** [comment]
Sure, but I don't see what I can do from my end to fix this :thinking:

**Comment by licaon-kter** [comment]
Ah, no, yes :) I meant to have it on our radar

**Comment by IzzySoft** [comment]
Maybe take a look at our recipe? Builds fine here and is RB. Though you'd need to update the recipe manually each time then as F-Droid lacks our automation. Build is what we call "flaky" (non-deterministic, probably thanks to some Google bug). Our builders can handle that.

**Comment by licaon-kter** [comment]
I'm not familiar with your paths exactly @IzzySoft 

Looking at https://android.izzysoft.de/repo/apk/eu.darken.sdmse I don't see a repro icon

Looking in https://github.com/obfusk/rbtlog/tree/log/recipes or https://github.com/obfusk/rbtlog/tree/log/logs I see no `eu.darken.sdmse`

**Comment by IzzySoft** [comment]
> I'm not familiar with your paths exactly

Yeah, I see – both times the wrong places :see_no_evil: First link should be:

https://apt.izzysoft.de/packages/eu.darken.sdmse

![image](https://github.com/user-attachments/assets/1fd359e2-ddf0-4b7a-ad4d-e3eb36a311b7)

There's your shield (the site you've linked to doesn't show RBs at all, that's just a secondary and simplified view on the repo). Klicking on that shield brings you to the builder that verified it (could be one or more builders, so you'd see one or more shields – yes, we have multiple builders). To bring you right to the corresponding line in the linked log:

https://codeberg.org/IzzyOnDroid/rbtlog/src/branch/izzy/log/logs/eu.darken.sdmse.json#L409

And the corresponding recipe would be [here](https://codeberg.org/IzzyOnDroid/rbtlog/src/branch/izzy/log/recipes/eu.darken.sdmse.yml#L259).

I know, visualization is lacking a bit at the moment. But that should hopefully change soon™ :wink:

**Comment by d4rken** [comment]
Can this be closed?

**Comment by licaon-kter** [comment]
1.3.5-rc1 has a Tag but no APK?

**Comment by d4rken** [comment]
I waited for the pipeline for Google Play to succeed, before approving the FOSS pipeline, but then went to sleep 😅 

https://github.com/d4rken-org/sdmaid-se/actions/runs/12836618905

**Comment by IzzySoft** [comment]
v1.3.6-rc0 seems to be RB no longer. I mean, our recipe produces the same APK 8 or 9 times in a row before the timeout kicks in, but – it's not yours. `baseline.prof` differs. Any idea what could cause this? Did something in the build workflow change?

**Comment by d4rken** [comment]
Between v1.3.5-rc1 and 1.3.6-rc0 there was nothing special in terms of code changes. 🤷

**Comment by IzzySoft** [comment]
Strange. You build with Github CI, right?

**Comment by linsui** [comment]
It's still RB in F-Droid CI. https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/8893937291

**Comment by licaon-kter** [comment]
it still took 2 tries

**Comment by IzzySoft** [comment]
Flaky builds are like playing lottery. It succeeded on our second builder today, at the 4th round.

**Comment by IzzySoft** [comment]
And baselines are bitches. v1.5.1-rc0 now failed 8 times in a row. Each round had the same hash, just not matching yours…

```diff
   num_type_ids=521
-  hot_method_region_size=5360
+  hot_method_region_size=5364
   dex_checksum=0xd57bae57
   num_method_ids=39890
 dex_data 0
-  num_hot_method_ids=1340
+  num_hot_method_ids=1341
   num_type_ids=521
   method_id=3442
   num_inline_caches=0
@@ -1728,6 +1728,8 @@
   num_inline_caches=0
   method_id=10432
   num_inline_caches=0
+  method_id=10447
+  num_inline_caches=0
   method_id=10448
```

Off by 1, as usual.
